### ISSUE TITLE
Automation view shift operation overwritten by extending clip length when previous automation data exists

### ISSUE THREAD

**Issue by entzmingerc** [SynthstromAudible/DelugeFirmware#1004]
### Please describe the problem:

(I was recommended to log this, but it's probably low priority.)

### Overwriting Rotate Operation
If I have automation data in the last 0.0.2 length section of a clip that's 1.0.2 long, then I decrease the length to 1.0.0 and rotate that automation, then I extend the clip back up to 1.0.2, the rotation of the automation is overwritten by the previous state before the rotation. It looks like the overwriting rotate doesn't do this while in instrument view rotating notes with automation shift OFF so that automation follows the rotation of the notes in the clip. 

### Stretching Behavior
Additionally, when extending the clip length past the point where no automation has been placed, Deluge stretches a very small segment of automation data across the new clip length region from the ending point of automation to the end of the clip. If this segment of automation data changes value (whether through linear interpolation or recorded knob movement) the change in value is stretched across the new clip length region. Often this is a constant value, but automation lets us make very fast changes in automation which I think exposed this existing behavior. 

### What is the expected behavior?

I think that if we rotate, it should forget automation within the clip at lengths outside of the current clip length, similar to how it works with notes in a clip (destructive). Or it should only rotate the specific section of 1.0.0 and leave the 0.0.2 section untouched so when the clip length is extended, the 0.0.2 section is the same as before and the 1.0.0 rotation still exists (non-destructive). Not sure. 

When extending the clip length, it's currently stretching a small segment of automation. I'm not sure if this is expected behavior or not. I was expecting that the last value of automation would be held constant as the clip length is extended. 

### Is there a relevant Pull request?

nah, c1.0.0

### What hardware did you reproduce it with?

7SEG

### What firmware did you reproduce it with?

Release 1.0/Amadeus

### What is the firmware name:

deluge-c1_0_0.bin

### If possible provide the steps to reproduce the issue and upload additional media:

Automation view, automation shift ON, linear interpolate ON  

1. Add in linear interpolation automation into the starting region of the clip
2. Rotate the automation to the left so that the starting automation wraps to the end of the clip
3. Increase clip length from 1.0.0 to 1.0.2. Should see it stretching if the smallest last rotated segment of automation has fast changing values (I had to zoom into 256th level).
4. Rotate the automation to the left so that the starting automation wraps to the end of the clip
5. Decrease clip length from 1.0.2 to 1.0.0 (now there is "hidden" automation in that 0.0.2 section)
6. Zoom out to look at the whole 1.0.0 section and rotate automation to the left. Observe the automation at the start is now at the end of the 1.0.0 size clip.
7. Increase the length again from 1.0.0 to 1.0.2. Error: When increasing the length again, the rotation performed in step 6 should now be overwritten by what the automation was after step 4. I think it should just be extending the last chunk of rotated automation at the end of the clip in step 6 instead of reverting to a previous state.
